### DISCUSSION TITLE
Allow forwarding id_token to backends for JWT Authentication

### DISCUSSION THREAD

**Discussion by albgus** [grafana/grafana#46240] [Data Sources]
Hello,

I’m trying to get Forward OAuth identity to work in order to authenticate people against data sources, using Azure AD as an identity provider. As a part of this I would like to use groups from Azure AD to restrict access to data sources that may contain sensitive information.

Enabling the option works well enough, each request to the data source contains an `Authorization: Bearer eyJ...` header, which is expected, I suppose. However, this token has the content `"aud": "00000003-0000-0000-c000-000000000000"` which apparently is a common “UUID” for Microsoft Graph, and importantly does not contain the groups claim.

There is also another header X-Id-Token that’s being sent to the backend. This has the audience key set to Client-ID of Grafana, as well as a groups key with all the users groups included. It seems that it should be possible to send this as a bearer token to a data source that can authenticate JWTs, and then have it restrict access based on the groups included in the token.

As far as I can tell there is no actual way to get Grafana to do this at the moment.

**Comment by akselleirv** [comment]
Hello @albgus,

Did you find a solution to this issue? I'm working on something similar where I need to check the groups a user is apart of, but I'm unable to do so as the ID Token is not sent to the data source.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
